scripts/resource/class_generator.py

```diff
@@ -392,136 +392,84 @@ def parse_explain(
-                            debug_content=debug_content,
-                            output_debug_file_path=output_debug_file_path,
-                            add_tests=add_tests,
-                        )
-                    )
+    if _fields := re.findall(r"  .*", raw_resource_dict[FIELDS_STR], re.DOTALL):
+        for line in [_line for _line in _fields[0].splitlines() if _line]:
+            if line.split()[0] in keys_to_ignore:
+                continue
 
-        if start_spec_field:
-            first_field_spec_found = True
-            field_spec_found = True
-            continue
-
-        if field_spec_found:
-            if not re.findall(rf"^{first_field_indent_str}\w", field):
-                if first_field_spec_found:
-                    resource_dict[SPEC_STR].append(
-                        get_arg_params(
-                            field=field,
-                            kind=kind,
-                            field_under_spec=True,
-                            debug=debug,
-                            debug_content=debug_content,
-                            output_debug_file_path=output_debug_file_path,
-                            add_tests=add_tests,
-                        )
+            # Process only top level fields with 2 spaces indent
+            if len(re.findall(r" +", line)[0]) == 2:
+                resource_dict[FIELDS_STR].append(
+                    get_arg_params(
+                        field=line,
+                        kind=kind,
+                        debug=debug,
+                        debug_content=debug_content,
+                        output_debug_file_path=output_debug_file_path,
+                        add_tests=add_tests,
                     )
+                )
 
-                    # Get top level keys inside spec indent, need to match only once.
-                    top_spec_indent = len(re.findall(r" +", field)[0])
-                    top_spec_indent_str = f"{' ' * top_spec_indent}"
-                    first_field_spec_found = False
-                    continue
-
-                if top_spec_indent_str:
-                    # Get only top level keys from inside spec
-                    if re.findall(rf"^{top_spec_indent_str}\w", field):
-                        resource_dict[SPEC_STR].append(
-                            get_arg_params(
-                                field=field,
-                                kind=kind,
-                                field_under_spec=True,
-                                debug=debug,
-                                debug_content=debug_content,
-                                output_debug_file_path=output_debug_file_path,
-                                add_tests=add_tests,
-                            )
-                        )
-                        continue
-
-            else:
-                break
-
+    # TODO: Do we have resources without `spec` or user configurable fields?
```

Comment:
do you mean like `ClusterOperator` which looks like this:
```
metadata....
  spec  <Object> -required-
  status        <Object>
...
```
?

Reply:
I mean resource that do not have to_dict, which the user cannot create, only get.

Reply:
so ClusterOperator is an example

Reply:
do you want to address it in this PR? add the name so we can address it later?

Reply:
I will do it in this PR